theseus_gui/src/store/instancesStore.js

```diff
@@ -0,0 +1,337 @@
+import { defineStore } from 'pinia'
+import { ofetch } from 'ofetch'
+import generated from '@/generated'
+
+export const useInstances = defineStore('instanceStore', {
+  state: () => ({
+    fakeInstances: [],
+    instances: [],
+    searchInput: '',
+    totalHits: 0,
+    currentPage: 1,
+    pageCount: 1,
+    offset: 0,
+    filter: 'Relevance',
+    categories: {
+      adventure: { label: 'Adventure', enabled: false },
+      cursed: { label: 'Cursed', enabled: false },
+      decoration: { label: 'Decoration', enabled: false },
+      economy: { label: 'Economy', enabled: false },
+      equipment: { label: 'Equipment', enabled: false },
+      food: { label: 'Food', enabled: false },
+      gameMechanics: { label: 'Game Mechanics', enabled: false },
+      library: { label: 'Library', enabled: false },
+      magic: { label: 'Magic', enabled: false },
+      management: { label: 'Management', enabled: false },
+      minigame: { label: 'Minigame', enabled: false },
+      mobs: { label: 'Mobs', enabled: false },
+      optimization: { label: 'Optimization', enabled: false },
+      social: { label: 'Social', enabled: false },
+      storage: { label: 'Storage', enabled: false },
+      technology: { label: 'Technology', enabled: false },
+      transportation: { label: 'Transportation', enabled: false },
+      utility: { label: 'Utility', enabled: false },
+      worldGeneration: { label: 'World Generation', enabled: false },
+    },
+    loaders: {
+      fabric: { label: 'Fabric', enabled: false },
+      forge: { label: 'Forge', enabled: false },
+      quilt: { label: 'Quilt', enabled: false },
+      rift: { label: 'Rift', enabled: false },
+      liteLoader: { label: 'LiteLoader', enabled: false },
+      risugami: { label: 'Risugami ModLoader', enabled: false },
+    },
```

Comment:
these shouldn't be hardcoded- since they are added dynamically via the database and can change depending on the project type. instead, they should be filtered depending on whatever project type is active in the browse menu (ex: modpack or mod)